### DISCUSSION TITLE
Docker

### DISCUSSION THREAD

**Discussion by Boganabi** [DrAlzahraniProjects/csusb_fall2024_cse6550_team1#29] [General]
Discuss things related to Docker here!

**Comment by ACraig7** [comment]
This error that Dr, Alzarani is getting seems to be related to an out of memory issue. The only solutions I have seen is to switch to conda cause it uses less memory but I dont think we can do that. 
![image](https://github.com/user-attachments/assets/26339456-bcd2-4782-a40e-0c1dd6c67667)

**Comment by itsnotmik** [comment]
@ACraig7 

It looks like we will need to downgrade our python version to 3.10 to work with the requirements of NeMo. Do you think we should look around for a docker image of mamba-forge that already includes that version of python? Or do you think we should manually downgrade and install 3.10? A new docker image would provide us faster build times but I am unsure how docker images are maintained and if one with 3.10 even exists

**Reply by itsnotmik** [reply]
I was unable to find any docker image for conda-forge that includes python 3.10. We should switch back to the python slim image and install mamba ourselves. 

As we discussed briefly, we believe that the error Dr. Alzahrani experienced was caused from the lack of `git pull origin main`. Since he was only cloning the repo (in the case that the repo existed) it was not being updated to the newest updates.

This should hold no weight on the other processes being worked on (UI and Jupyter) and should be easy to integrate

**Reply by ACraig7** [reply]
Yeah, I don't think that exists. If we are going to back to 3.10 base image we should implement code similar to team 3, which we had before the current implementation. I will work on getting a 3.10 slim branch going.

**Reply by itsnotmik** [reply]
Okay sounds good, are you currently going to work on it? I am free for about 1 and a half hours so I can make the initial changes if you are not free currently

**Reply by ACraig7** [reply]
Testing it now. I will push a new branch called "3.10" or something like that in a minute.

**Reply by ACraig7** [reply]
@itsnotmik branch 3.10 is published

**Comment by Sai0235** [comment]
I want to work on /team1 url but i need ngix setup in docker file can anyone help me

**Reply by ACraig7** [reply]
if you look at team 4 repo, they seem to have it set up. https://github.com/DrAlzahraniProjects/csusb_fall2024_cse6550_team4

**Reply by itsnotmik** [reply]
There is currently a pull request which contains a base working version of /team1 URL

Also look at branch [URL_redirect_using_nginx](https://github.com/DrAlzahraniProjects/csusb_fall2024_cse6550_team1/tree/URL_redirect_using_nginx)

Currently it does not work on the URL `localhost/team1` but does work on `localhost/team1/`
